fedimint-core/src/encoding/mod.rs

```diff
@@ -345,209 +343,6 @@ impl_encode_decode_tuple!(T1, T2);
-            let len = u64::consensus_decode_from_finite_reader(d, &ModuleRegistry::default())?;
-
-            let mut len: usize =
-                usize::try_from(len).map_err(|_| DecodeError::from_str("size exceeds memory"))?;
-
-            let mut bytes = vec![];
-
-            // Adapted from <https://github.com/rust-bitcoin/rust-bitcoin/blob/e2b9555070d9357fb552e56085fb6fb3f0274560/bitcoin/src/consensus/encode.rs#L667-L674>
-            while len > 0 {
-                let chunk_start = bytes.len();
-                let chunk_size = core::cmp::min(len, CHUNK_SIZE);
-                let chunk_end = chunk_start + chunk_size;
-                bytes.resize(chunk_end, 0u8);
-                d.read_exact(&mut bytes[chunk_start..chunk_end])
-                    .map_err(DecodeError::from_err)?;
-                len -= chunk_size;
-            }
-
-            // unsafe: we've just checked that T is `u8` so the transmute here is a no-op
-            return Ok(unsafe { mem::transmute::<Vec<u8>, Self>(bytes) });
-        }
-        let len = u64::consensus_decode_from_finite_reader(d, modules)?;
-
-        // `collect` under the hood uses `FromIter::from_iter`, which can potentially be
-        // backed by code like:
-        // <https://github.com/rust-lang/rust/blob/fe03b46ee4688a99d7155b4f9dcd875b6903952d/library/alloc/src/vec/spec_from_iter_nested.rs#L31>
-        // This can take `size_hint` from input iterator and pre-allocate memory
-        // upfront with `Vec::with_capacity`. Because of that untrusted `len`
-        // should not be used directly.
-        let cap_len = cmp::min(8_000 / mem::size_of::<T>() as u64, len);
-
-        // Up to a cap, use the (potentially specialized for better perf in stdlib)
-        // `from_iter`.
-        let mut v: Self = (0..cap_len)
-            .map(|_| T::consensus_decode_from_finite_reader(d, modules))
-            .collect::<Result<Self, DecodeError>>()?;
-
-        // Add any excess manually avoiding any surprises.
-        while (v.len() as u64) < len {
-            v.push(T::consensus_decode_from_finite_reader(d, modules)?);
-        }
-
-        assert_eq!(v.len() as u64, len);
-
-        Ok(v)
-    }
-}
-
-impl<T> Encodable for VecDeque<T>
-where
-    T: Encodable + 'static,
-{
-    fn consensus_encode<W: std::io::Write>(&self, writer: &mut W) -> Result<usize, Error> {
-        let mut len = (self.len() as u64).consensus_encode(writer)?;
-        for i in self {
-            len += i.consensus_encode(writer)?;
-        }
-        Ok(len)
-    }
-}
-
-#[test]
-fn vec_decode_sanity() {
-    let buf = [
-        0xffu8, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-    ];
-
-    // On malicious large len, return an error instead of panicking.
-    assert!(Vec::<u8>::consensus_decode(&mut buf.as_slice(), &ModuleRegistry::default()).is_err());
-    assert!(Vec::<u16>::consensus_decode(&mut buf.as_slice(), &ModuleRegistry::default()).is_err());
-}
```

Comment:
Note for reviewer: This test wasn't actually testing malicious lengths, it was testing a length prefix that is greater than the number of decodable elements. This test was replaced by `test_lists`.